src/xmlutil.rs

```diff
@@ -19,12 +19,47 @@ impl XmlParseError {
 /// syntactic sugar for the XML event stack we pass around
 pub type XmlStack<'a> = Peekable<Events<'a, Response>>;
 
+pub trait Peek {
+    fn peek(&mut self) -> Option<&XmlEvent>;
+}
+
+pub trait Next {
+	fn next(&mut self) -> Option<XmlEvent>;
+}
+
+// Wraps the Hyper Response type
+pub struct XmlResponseFromAws<'b> {
+	xml_stack: Peekable<Events<'b, Response>> // refactor to use XmlStack type?
+}
+
+// I cannot explain how these lifetimes work to a child, therefore I need to understand them better:
```

Comment:
I believe 'b here refers to the lifetime of the memory that contains the raw text that Events is parsing XmlEvents from.  It's worth keeping an eye on so we don't accidentally copy the whole XML document around unnecessarily (I already fixed that duplication once a long time ago)